src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -156,8 +157,12 @@ uct_ib_mlx5_devx_reg_ksm_data_mt(uct_ib_mlx5_md_t *md, void *address,
                                  uct_ib_mlx5_devx_ksm_data_t *ksm_data,
                                  struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
 {
-    void *mr_address = address;
-    size_t list_size = ksm_data->mr_num;
+    size_t chunk_size     = md->super.config.mt_reg_chunk;
+    intptr_t chunk_prefix = (intptr_t)address % chunk_size;
+    void* mr_address      = UCS_PTR_BYTE_OFFSET(address, -chunk_prefix);
+    uint64_t ksm_iova     = iova - chunk_prefix;
+    size_t length         = ksm_data->length + chunk_prefix;
```

Comment:
do we really have to "duplicate" this logic here in addition to what we have in uct_ib_mlx5_devx_reg_mt?
maybe pass aligned address and iova from uct_ib_mlx5_devx_reg_mt to this function, instead of also doing the padding/alignment here?

Reply:
Done in accordance to offline discussion